### PR TITLE
Slice out "rules" for elements not found in system

### PR DESCRIPTION
**Pull request by mattwthompson**
Added a pre-processing step in ```find_atomtypes``` that internally throws out rules based on atomtypes not found in the system and reduces the size of the ```rules``` dictionary accordingly. For example, if you were atomtyping alkanes with OPLS, it would remove rules considering N, O, F, etc. before atomtyping. This does not, however, modify the ```Forcefield``` itself because the user may wish to re-use that forcefield for a system that contains a different set of unique elements.

### CODE REVIEW

File: foyer/atomtyper.py [hunk 1]
```diff
@@ -18,6 +18,27 @@ def find_atomtypes(topology, forcefield, max_iter=10):
...
+                element = atom.select('atom_symbol').strees[0].tail[0]
+            except IndexError:
+                try:
+                    atomic_num = atom.select('atomic_num').strees[0].tail[0]
+                    element = pt.Element[int(atomic_num)]
+                except IndexError:
+                    element = None
+        else:
+            element = None
+        if element in system_elements:
```
Comment (ctk3b) [inline foyer/atomtyper.py]: I believe this still tosses out rules whose element resolves to `None`.

I think either this line should be:

`if element is None or element in system_elements`

or you just add `None` to `system_elements` before the loop

**Comment by ctk3b** [comment]
Excellent, thank you. You tested this on a few systems right? What are the ballpark speedups/system specs?

**Comment by mattwthompson** [comment]
I was messing around with my ILs earlier, but here I did a benchmark of 200 hexane molecules using the existing OPLS force field.

Using the existing master branch, atomtyping takes 55 seconds, and on this branch it takes 48 seconds. For heaxane, it drops the number of rules from 89 to 67. However, if I turn all of the hydrogens into fluorines, it is cut down to 48 rules and the comparison is 30 to 23 seconds. So it may typically be 10-20% but it probably varies with the size of the original force field and the chemistry of the system.

**Comment by chrisiacovella** [comment]
Can you do a quick test as a function of system size (e.g., 2000, 20000 hexanes)?  Does this scale with system-size, as one might expect.

**Pull request merged**